lib/linked_list.rb

```diff
@@ -12,148 +11,355 @@ def initialize(value, next_node = nil)
 
 # Defines the singly linked list
 class LinkedList
-    def initialize
-      @head = nil # keep the head private. Not accessible outside this class
-    end
+  def initialize
+    @head = nil # keep the head private. Not accessible outside this class
+  end
+
+  # method to add a new node with the specific data value in the linked list
+  # insert the new node at the beginning of the linked list
+  # Time Complexity: O(1)
+  # >> Constant num of actions - create new node,
+  # set next to old head, set head pointer to new node
+  # Space Complexity: O(1)
+  # >> Always adding 1 new node
+  def add_first(value)
+    new_node = Node.new(value)
+
+    new_node.next = @head
+    @head = new_node

```

Comment:
This is a little more condense.

```suggestion
  @head = Node.new(value, @head)
```

Reply:
Oh yes! This is great